src/ParseProxyValues.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Primitives;
+
+namespace AspNetCore.ForwardedHttp
+{
+    public class ForwardedValues
+    {
+        public StringSegment For { get; set; }
+        public StringSegment By { get; set; }
+        public StringSegment Host { get; set; }
+        public StringSegment Proto { get; set; }
+    }
+
+    public class ParseProxyValues
+    {
+        private static readonly char[] CommaChar = new[] { ',' };
+        private static readonly char[] SemiColonChar = new[] { ';' };
+
+        public ParseProxyValues()
+        {
+
+        }
+
+        public static bool TryParse(StringValues values, out ParseProxyValues parseProxyValues)
```

Comment:
Not sure what you get from returning this type instance. The whole class could be static.
```suggestion
        public static bool TryParse(StringValues values, out IList<ForwardedValues> parseProxyValues)
```